src/core_plugins/kibana/public/management/sections/indices/source_filters/source_filters.js

```diff
@@ -92,12 +92,20 @@ uiModules.get('kibana')
       }
 
       delete(filter) {
-        if (this.editing === filter) {
-          this.editing = null;
-        }
+        const doDelete = () => {
+          if (this.editing === filter) {
+            this.editing = null;
+          }
 
-        this.$scope.indexPattern.sourceFilters = without(this.all(), filter);
-        return this.save();
+          this.$scope.indexPattern.sourceFilters = without(this.all(), filter);
+          return this.save();
+        };
+
+        const confirmModalOptions = {
+          confirmButtonText: 'Delete filter',
+          onConfirm: doDelete
+        };
+        confirmModal(`Are you sure want to delete the filter?`, confirmModalOptions);
```

Comment:
Small nit: can we change this to "Are you sure you want to delete this filter", instead of *the* filter? This is how other confirmation messages refer to an anonymous object and it reads a little more clearly to me.